ProcessLib/HydroMechanics/HydroMechanicsProcess-impl.h

```diff
@@ -84,101 +153,109 @@ void HydroMechanicsProcess<DisplacementDim>::initializeConcreteProcess(
     MeshLib::Mesh const& mesh,
     unsigned const integration_order)
 {
+    const int mechinical_process_id = _use_monolithic_scheme ? 0 : 1;
+    const int deformation_variable_id = _use_monolithic_scheme ? 1 : 0;
     ProcessLib::HydroMechanics::createLocalAssemblers<
         DisplacementDim, HydroMechanicsLocalAssembler>(
         mesh.getDimension(), mesh.getElements(), dof_table,
         // use displacment process variable for shapefunction order
-        getProcessVariables()[1].get().getShapeFunctionOrder(),
+        getProcessVariables(mechinical_process_id)[deformation_variable_id]
+            .get()
```

Comment:
Use dereference operator instead of get(). A unique pointer is a pointer too with all the related semantics.